### PR TITLE
feat: use resource selector for playbook configs

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: db/playbooks.go [hunk 1]
```diff
@@ -103,28 +109,52 @@ func FindPlaybooksForCheck(ctx context.Context, checkType string, tags map[strin
...
+	for _, pb := range playbooks {
+		var spec v1.PlaybookSpec
+		if err := json.Unmarshal(pb.Spec, &spec); err != nil {
+			return nil, fmt.Errorf("error unmarshaling playbook spec: %w", err)
+		}
 
-	var playbooks []api.PlaybookListItem
-	err := query.Model(&models.Playbook{}).Find(&playbooks).Error
-	return playbooks, err
+		configIDs, err := query.FindConfigIDsByResourceSelector(ctx, spec.Configs...)
```
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply db/playbooks.go:133]: yeah, but we want to cache the entire thing and use cache for most of them

Using per id would prevent caching

File: db/playbooks.go [hunk 2]
```diff
@@ -103,28 +110,58 @@ func FindPlaybooksForCheck(ctx context.Context, checkType string, tags map[strin
...
+	logger.Infof("YASH 2")
+	if err := ctx.DB().Model(&models.Playbook{}).Where("spec->>'configs' IS NOT NULL").Where("deleted_at IS NULL").Find(&playbooks).Error; err != nil {
+		return nil, fmt.Errorf("error finding playbooks with configs: %w", err)
 	}
 
-	query := ctx.DB().
-		Select("DISTINCT ON (playbooks.id) playbooks.id, playbooks.name, playbooks.spec->'parameters' as parameters").
-		Joins(joinQuery, joinArgs...)
+	configIDPlaybooks := make(map[string][]string)
+	logger.Infof("YASH 3")
```
[the remark this answers is not in the document]
Comment (yashmehrotra) [reply db/playbooks.go]: Fixed, my bad

File: db/playbooks.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (moshloop) [reply db/playbooks.go:133]: Can't we just add go code for matching i.g. ResourceSelector.matches(config) leveraging https://github.com/kubernetes/apimachinery/blob/master/pkg/labels/selector.go

[the remark this answers is not in the document]
Comment (yashmehrotra) [reply db/playbooks.go]: Done

**Pull request merged**